packages/server/lib/util/graceful_crash_handling.ts

```diff
@@ -0,0 +1,158 @@
+import type { ProjectBase } from '../project-base'
+import * as errors from '../errors'
+import Debug from 'debug'
+
+const debug = Debug('cypress:util:crash_handling')
+
+interface ReporterTestAttempt {
```

Comment:
This isn't the correct place for these typedefs - but Reporter is not converted to ts yet